ckan/lib/helpers.py

```diff
@@ -289,6 +289,18 @@ def _get_auto_flask_context():
             return _cli_test_request_context
 
 
+@core_helper
+def default_url_for(package_type, **kw):
+    '''Return the URL for an endpoint given some parameters, or use the default '/dataset' url if unable.
```

Comment:
Might want to add more detail about why this helper exists, why it's useful.

Reply:
nice idea on trying to solve the issue. only problem is that you can't default to read and its not tied into url_for below.  

Is there a method to verify if a controler.function exists. if so that could be made as a helper and then you can wrap the delete system in 'if (verify xys.read exists) else (dataset.read) etc

Reply:
> nice idea on trying to solve the issue. only problem is that you can't default to read and its not tied into url_for below.
> 
> Is there a method to verify if a controler.function exists. if so that could be made as a helper and then you can wrap the delete system in 'if (verify xys.read exists) else (dataset.read) etc

I've tried changing the helper function to work for all urls, not just read, and let it use the url_for function name, though I'm not completely sure this is right.
And I tried searching but I couldn't find a method that could verify if a controller function exists.